### ISSUE TITLE
Too-short urls break registerUrl--extra byte padding.

### ISSUE THREAD

**Issue by Smithgift** [ethereum/go-ethereum#1293]
A short url (such as "file:///tmp/info.json" as suggested in the tutorial) will end up with %00s padding it in the registrar, which then breaks contractInfo.get().

**Comment by ethers** [comment]
How do you check the registrar and see the %00s padding ?

I always get 0 when calling `get()` even with using the sample contract in the wiki
```
> {ster(primaryAccount, address, contract, "/tmp/averylongfilenameoverheresomemuchlongerfilenametoavoidbug.json")
'0x4b7993849ee446cedfc883adba4246426f3a0b7ad8cdf9cd89b2d28a1609e10f'
> admin.contractInfo.registerUrl(primaryAccount, hash, bigUrl)
true
> 
> hash
'0x4b7993849ee446cedfc883adba4246426f3a0b7ad8cdf9cd89b2d28a1609e10f'
> bigUrl
'file:///tmp/averylongfilenameoverheresomemuchlongerfilenametoavoidbug.json'
> address
'0xa2eb0dd8342dd5100e0a4fffc530a291aeb78d7d'
> admin.contractInfo.get(address)
GetURLhint: URL hint not found for '0x0000000000000000000000000000000000000000000000000000000000000000'
undefined
> contract
{
  code: '0x6060604052606d8060116000396000f30060606040526000357c010000000000000000000000000000000000000000000000000000000090048063c6888fa1146037576035565b005b6046600480359060200150605c565b6040518082815260200191505060405180910390f35b60006007820290506068565b91905056',
  info: {
    abiDefinition: [{
      constant: false,
      inputs: [{
        name: 'a',
        type: 'uint256'
      } ],
      name: 'multiply',
      outputs: [{
        name: 'd',
        type: 'uint256'
      } ],
      type: 'function'
    } ],
    compilerVersion: '0.9.26',
    developerDoc: {
      methods: {
      }
    },
    language: 'Solidity',
    languageVersion: '0',
    source: 'contract test { function multiply(uint a) returns(uint d) { return a * 7; } }',
    userDoc: {
      methods: {
      }
    }
  }
}
```

**Comment by Smithgift** [comment]
I've only tried this on a private chain. Are you on a private chain, too? If so, have you used admin.contractInfo.newRegistry()? I believe you have to create the registry on the same session as using the NatSpec, as I don't think it saves the registry's new address.

**Comment by ethers** [comment]
Thanks, I'm using the Olympic chain

**Comment by zelig** [comment]
fixed in frozen #1429
